app/webpack/reactjs/components/admin/tests/welcome.tsx

```diff
@@ -1,11 +1,14 @@
 import * as React from 'react';
 
-interface WelcomeProps {
-  message: string
+type WelcomeProps = {
```

Comment:
Is there a reason to favor a `type`?  in our current project i've been leaning on interface and in my readings I'm not sure if there is a fundamental benefit to one or the other.  Just curious why the switch here?

Reply:
I read it briefly [here](https://github.com/typescript-cheatsheets/react/blob/main/README.md#types-or-interfaces). That's why I switched it. However as I'm doing more research, I'm not sure if it's a strong case to choose `type`. There were three reasons listed for preference of `type` in React from [this article](https://medium.com/@martin_hotell/interface-vs-type-alias-in-typescript-2-7-2a8f1777af4c)(screenshot attached), as recommended reading from the cheatsheet. In my personal opinion, I think the first two are not significant as long as devs are consistent within a codebase. I don't think the third reason is as important either since we favor composition nowadays in React. 

As what the cheat sheet said about `type` is better for union types, I'm also questioning if that's true. It doesn't seem to me that `interface` does worse on unions. But I could be missing something here. 

So my conclusion is the cheatsheet and official handbook are out of date...

I've also found a more up-to-date article [here](https://pawelgrzybek.com/typescript-interface-vs-type/). But I still cannot see reasons to favor one over another, for general cases in React, unless declaration merging or mapped types happens a lot in real-life typescript-react projects. However, `interface` might sound more correct conceptually with prop types. 

So I guess I don't have a strong preference after all. 

<img width="968" alt="Screen Shot 2020-11-20 at 3 41 14 PM" src="https://user-images.githubusercontent.com/6734092/99928680-24839400-2d18-11eb-9ddd-590ad547f157.png">

Reply:
I'm fine with types.  Certainly I think with unions (and I guess intersections?) I might like `type` better because you can
```
type ThisThing = ThatThing | TheOtherThing;
```
which I don't know how I'd do with an interface.  And this is something (on my current project) we do occasionally like
```
type ButtonType = "big" | "small"
type OtherElementSize = "medium" | "small"
type ElementSize = ButtonType | OtherElementSize
```

not a great example, but maybe it shows what I'm thinking.  I also think our current project is a mess and the fact that you're reading (which I'm not) I'm happy to lean on your direction.  I'm excited to see this stuff work nicely because I think the current project has become a mess because it was ill conceived early leading to bad patterns that are hard to unwind or change.  Maybe we can do better cuz we're smarter and moving very slowly :D 

So ... let's go with `type` for props.

Reply:
incidentally - the first argument on that page makes me wonder.  "it's shorter to type" is not super compelling given autocomplete and all the other stupid IDE helpers we have. :D. but I'm still good with the rest of the argument.

Reply:
Yea that's a good point. `interface` cannot be assigned. OKies! `type` it is!